gui/src/app/SPAnalysis/SPAnalysisQueryLoading.ts

```diff
@@ -0,0 +1,113 @@
+import { SPAnalysisDataModel, initialDataModel } from "./SPAnalysisDataModel";
+
+
+enum QueryParamKeys {
+    StanFile = "stan",
+    DataFile = "data",
+    SamplingOpts = "sampling_opts",
+    Title = "title",
+    SONumChains = 'num_chains',
+    SONumWarmup = 'num_warmup',
+    SONumSamples = 'num_samples',
+    SOInitRadius = 'init_radius',
+    SOSeed = 'seed'
+}
+
+type QueryParams = {
+    [key in QueryParamKeys]: string | null
+}
+
+export const fromSearchParams = (searchParams: URLSearchParams) => {
+    for (const key of searchParams.keys()) {
+        // warn on unknown keys
+        if (!Object.values(QueryParamKeys).includes(key as QueryParamKeys)) {
+            console.warn('Unknown query parameter', key)
+        }
+    }
+
+    const query: QueryParams = {
+        stan: searchParams.get(QueryParamKeys.StanFile),
+        data: searchParams.get(QueryParamKeys.DataFile),
+        sampling_opts: searchParams.get(QueryParamKeys.SamplingOpts),
+        title: searchParams.get(QueryParamKeys.Title),
+        num_chains: searchParams.get(QueryParamKeys.SONumChains),
+        num_warmup: searchParams.get(QueryParamKeys.SONumWarmup),
+        num_samples: searchParams.get(QueryParamKeys.SONumSamples),
+        init_radius: searchParams.get(QueryParamKeys.SOInitRadius),
+        seed: searchParams.get(QueryParamKeys.SOSeed),
+    }
+    return query
+
+}
+
+const tryFetch = async (url: string) => {
+    console.log('Fetching content from', url)
+    try {
+        const req = await fetch(url)
+        if (!req.ok) {
+            console.error('Failed to fetch from url', url, req.status, req.statusText)
+            return undefined
+        }
+        return await req.text()
+    } catch (err) {
+        console.error('Failed to fetch from url', url, err)
+        return undefined
+    }
+}
+
+const deepCopy = (obj: any) => {
+    return JSON.parse(JSON.stringify(obj))
+}
+
+export const fetchRemoteAnalysis = async (query: QueryParams) => {
+
+    // any special 'project' query could be loaded here at the top
+    const data: SPAnalysisDataModel = deepCopy(initialDataModel)
+
+    if (query.stan) {
```

Comment:
Noting several `await`s here--there's always an asterisk in my head to want to make sure we aren't demanding that these be serialized when the fetches could be happening in parallel.

From a *very brief* perusal I wonder if we actually want to defer the `await`s until after all the `fetch`s have been triggered?
...sorry, that's a little gnomic, let me know if you'd like me to expand on that thought.

Reply:
Is 22aaba7fe72c41ba06a3627535ead1c11a64d142 what you meant? This isn't something I had thought about before but it makes sense

Reply:
Yeah, this seems more like.